File: priv/combined_metrics/samples/code_smells/no_debug_print_statements/bad/payment.py
class Payment:
    def charge(self, user, amount, card):
        print("charging user:", user.id)
        print("card details:", card)
        print("amount:", amount)

        validated = self._validate_card(card)
        if not validated:
            print("card validation failed")
            return None

        print("card validated successfully")
        result = self._call_gateway(validated, amount)
        print("gateway result:", result)
        return result

    def refund(self, transaction_id, amount):
        print("starting refund for transaction:", transaction_id)
        transaction = self._fetch_transaction(transaction_id)
        print("found transaction:", transaction)

        if transaction["amount"] < amount:
            print("refund amount exceeds original")
            return {"error": "exceeds_original"}

        print("processing refund of", amount)
        result = self._call_refund_api(transaction, amount)
        print("refund result:", result)
        return result

    def calculate_fee(self, amount, method):
        print("fee calc input:", amount, method)
        if method == "credit_card":
            fee = amount * 0.029 + 0.30
        elif method == "debit_card":
            fee = amount * 0.015
        else:
            fee = amount * 0.035
        print("calculated fee:", fee)
        return fee

    def _validate_card(self, card):
        return card

    def _call_gateway(self, _card, _amount):
        return {"transaction_id": "txn_123"}

    def _fetch_transaction(self, _id):
        return {"amount": 100.0}

    def _call_refund_api(self, _transaction, _amount):
        return {"status": "refunded"}

File: priv/combined_metrics/samples/code_smells/no_debug_print_statements/bad/payment.js
export class Payment {
  charge(user, amount, card) {
    console.log('charging user:', user.id);
    console.log('card details:', card);
    console.log('amount:', amount);

    const validated = this.validateCard(card);
    if (!validated) {
      console.log('card validation failed');
      return null;
    }

    console.log('card validated successfully');
    const result = this.callGateway(validated, amount);
    console.log('gateway result:', result);
    return result;
  }

  refund(transactionId, amount) {
    console.log('starting refund for transaction:', transactionId);
    const transaction = this.fetchTransaction(transactionId);
    console.log('found transaction:', transaction);

    if (transaction.amount < amount) {
      console.log('refund amount exceeds original');
      return { error: 'exceeds_original' };
    }

    console.log('processing refund of', amount);
    const result = this.callRefundApi(transaction, amount);
    console.log('refund result:', result);
    return result;
  }

  calculateFee(amount, method) {
    console.log('fee calc input:', amount, method);
    let fee;
    switch (method) {
      case 'credit_card':
        fee = amount * 0.029 + 0.3;
        break;
      case 'debit_card':
        fee = amount * 0.015;
        break;
      default:
        fee = amount * 0.035;
    }
    console.log('calculated fee:', fee);
    return fee;
  }

  validateCard(card) {
    return card;
  }

  callGateway(_card, _amount) {
    return { transactionId: 'txn_123' };
  }

  fetchTransaction(_id) {
    return { amount: 100.0 };
  }

  callRefundApi(_transaction, _amount) {
    return { status: 'refunded' };
  }
}
